### PR TITLE
connect: delete and consolidate more code

### PR DESCRIPTION
**Pull request by kyessenov**
List of changes:
* delete dead or duplicated code;
* report different server name header for waypoints;
* **major**: disable forwarding XFCC, that doesn't work for CONNECT termination;
* **major**: require client certificates for mTLS in CONNECT termination listener.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
/test integ-security-multicluster

This test seems fail every where

File: pilot/pkg/networking/core/v1alpha3/cluster_waypoint.go [hunk 1]
```diff
@@ -62,7 +75,7 @@ func (configgen *ConfigGeneratorImpl) buildWaypointInboundClusters(
 	clusters := make([]*cluster.Cluster, 0)
 	// Creates "main_internal" cluster to route to the main internal listener.
 	// Creates "encap" listener to route to the encap listener.
-	clusters = append(clusters, cb.buildWaypointInboundInternal()...)
+	clusters = append(clusters, MainInternalCluster, EncapCluster)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_waypoint.go:78]: L77 comment is encap cluster not listener

**Review by hzxuzhonghu — APPROVED** [review]
BTW, one unrelated question why TLS1.3 is enforced for HBONE?

**Comment by kyessenov** [comment]
@hzxuzhonghu TLS 1.3 has less options - no cipher suites/etc. It does have a problem with "early data" but that doesn't affect application-level ACKs with HTTP CONNECT. Essentially, TLS 1.3 doesn't wait for server to ACK so any early data sent is dropped if server denies the client (m in mTLS). But HTTP CONNECT waits for 200 before sending TCP payload so it doesn't suffer from this.

**Pull request merged**